### PR TITLE
Maintenance: XCode14 migration

### PR DESCRIPTION
**Pull request by wutschel**
## Description

Closes https://github.com/xbmc/Official-Kodi-Remote-iOS/issues/716.
Closes  https://github.com/xbmc/Official-Kodi-Remote-iOS/issues/720.

This PR applies changes to make the Remote App work when compiled with XCode 14. 

**Toolbar layout**
The toolbar will not extend into safe area anymore. Instead, the toolbars are now made transparent and a background view with the desired color is added. For the NowPlaying screen the colorization of the bottom toolbar is removed, now being consistent with the navigation bar. In addition, the playlist toolbar is now faded in/out instead of vertical movement.

**Navigation bar title**
A workaround is required to show `self.navigationItem.title` properly when entering a menu the first time.

## Summary for release notes

Maintenance: XCode14 migration
Bugfix: Resolves layout issue around iPhone 14 Pro island

### CODE REVIEW

**Comment by kambala-decapitator** [comment]
FYI once we switch to Xcode 14, we must raise deployment target to 11.0 (dropping armv7 support). This should be done only when we are sure that release is basically flawless as users with iOS 9/10 will have no way to get fixes (other than building from source).

One more option would be producing 2 builds: one with the latest toolchain that goes to appstore, another one with Xcode 13 that will only be attached to GitHub releases.

**Comment by wutschel** [comment]
Yep, we should leave the users with a proper working version. We could even think of raising the iOS deployment target _now_, version 1.10.1 works pretty well since 5+ months. I just checked the AppStore metrics and we have <1% sessions using iOS 9 or 10 (6 months: about 8.2k out of 1+ million sessions).

**Comment by kambala-decapitator** [comment]
based on release notes I'd rather make upcoming release available to iOS 9 as well

**Comment by wutschel** [comment]
Ok for me. I have no strong opinion on this.

**Comment by kambala-decapitator** [comment]
please also raise deployment target in this PR. You might also want to check/fix new warnings after that.

**Comment by wutschel** [comment]
> please also raise deployment target in this PR. You might also want to check/fix new warnings after that.

Done. No new warnings, I am ahead of this already with the work of the last months.

**Comment by kambala-decapitator** [comment]
received this on submission, however Apple news don't have it...

> ITMS-90725: SDK Version Issue - This app was built with the iOS 15.5 SDK. Starting April 2023, all iOS apps submitted to the App Store must be built with the iOS 16.1 SDK or later, included in Xcode 14.1 or later.

**Comment by wutschel** [comment]
> received this on submission, however Apple news don't have it...
> 
> > ITMS-90725: SDK Version Issue - This app was built with the iOS 15.5 SDK. Starting April 2023, all iOS apps submitted to the App Store must be built with the iOS 16.1 SDK or later, included in Xcode 14.1 or later.

Then let‘s bring this in now to gain some test depth. Good to have this one in a dedicated build.

**Comment by kambala-decapitator** [comment]
https://developer.apple.com/news/?id=jd9wcyov

> Starting April 25, 2023, iOS, iPadOS, and watchOS apps submitted to the App Store must be built with Xcode 14.1 or later

**Comment by wutschel** [comment]
> Starting April 25, 2023, iOS, iPadOS, and watchOS apps submitted to the App Store must be built with Xcode 14.1 or later

There we go. I used XCode 14.2 to verify the changes in this PR. When do plan to migrate your environment?

**Comment by wutschel** [comment]
We are on similar environments then. :) 
This change can of course also go in right now. It is just making changes which will later be required by XCode 14.x.

**Comment by wutschel** [comment]
> I have 14.2. Plan to upgrade to Ventura (and 14.3) soon actually.

Let me know once you upgraded to 14.3. I will do the same then to have an identical environment in case of debugging or reproduction should be required.

**Comment by kambala-decapitator** [comment]
> > I have 14.2. Plan to upgrade to Ventura (and 14.3) soon actually.
> 
> Let me know once you upgraded to 14.3. I will do the same then to have an identical environment in case of debugging or reproduction should be required.

will do end of April, once submissions with Xcode 13 are no longer possible (might still need it soon)

File: XBMC Remote/NowPlaying.m [hunk 1]
```diff
@@ -2628,6 +2628,13 @@ - (void)viewDidLoad {
     playlistTableView.contentInset = UIEdgeInsetsMake(0, 0, 44, 0);
     self.edgesForExtendedLayout = UIRectEdgeNone;
     
+    // Transparent toolbar
+    [playlistToolbar setBackgroundImage:[UIImage new]
+                     forToolbarPosition:UIBarPositionAny
+                             barMetrics:UIBarMetricsDefault];
+    [playlistToolbar setShadowImage:[UIImage new]
+                 forToolbarPosition:UIBarPositionAny];
```
Comment (kambala-decapitator) [inline XBMC Remote/NowPlaying.m:2609]: I suggest to extract this to toolbar extension / helper method to avoid copy-paste

**Review by kambala-decapitator — APPROVED** [review]

File: XBMC Remote/NowPlaying.m (hunk 1 above)
Comment (wutschel) [reply XBMC Remote/NowPlaying.m]: Good idea, added a fixup commit. Will squash the first two and this fixup into a single commit.

**Comment by wutschel** [comment]
Oh, just saw you approved this one already. I added [97f90fe](https://github.com/xbmc/Official-Kodi-Remote-iOS/pull/727/commits/97f90feb0fc4ce10afbd6dd1233a5bd71a39a9b0) as well, is this ok for you as well? If not, I will simply remove it.

**Review by kambala-decapitator — APPROVED** [review]
> I added [97f90fe](https://github.com/xbmc/Official-Kodi-Remote-iOS/pull/727/commits/97f90feb0fc4ce10afbd6dd1233a5bd71a39a9b0) as well, is this ok for you as well?

sure, makes complete sense

**Comment by wutschel** [comment]
Squashed the fixup with the first two commits (transparent toolbars) and force-pushed.

**Pull request merged**